modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -2095,6 +2098,23 @@ private Expression<?> _getFunctionExpression(
 				relatedObjectDefinition.getObjectDefinitionId(),
 				GetterUtil.getString(
 					objectFieldSettingsValues.get("objectFieldName")));
+
+			if (objectRelationshipSelf) {
+				if (column.getTable(
+					).getTableName(
+					).equals(
+						relatedDynamicObjectDefinitionTable.getTableName()
+					)) {
+
+					column = relatedDynamicObjectDefinitionTable.getColumn(
+						column.getName());
+				}
+				else {
+					column =
+						relatedExtensionDynamicObjectDefinitionTable.getColumn(
+							column.getName());
+				}
+			}
```

Comment:
The bug is happening because the table names of the subquery and the main query are the same as it is a self-relationship. To solve this I put an alias in the subquery table. But this method here 
`column = _objectFieldLocalService.getColumn(
				relatedObjectDefinition.getObjectDefinitionId(),
				GetterUtil.getString(
					objectFieldSettingsValues.get("objectFieldName")));
`
doesn't use the dynamics tables already created here : 
`DynamicObjectDefinitionTable
					relatedDynamicObjectDefinitionTable =
						new DynamicObjectDefinitionTable(
							relatedObjectDefinition,
							_objectFieldLocalService.getObjectFields(
								relatedObjectDefinition.
									getObjectDefinitionId()),
							relatedObjectDefinition.getDBTableName());
				DynamicObjectDefinitionTable
					relatedExtensionDynamicObjectDefinitionTable =
						new DynamicObjectDefinitionTable(
							relatedObjectDefinition,
							_objectFieldLocalService.getObjectFields(
								relatedObjectDefinition.
									getObjectDefinitionId()),
							relatedObjectDefinition.getExtensionDBTableName());`  
so the column is created without an alias. So I need to use the `getColum` of the previously created dynamics tables.